### PR TITLE
Resolve THIRD-PARTY license issues [HZ-2159, HZ-2774][5.3.z]

### PR DESCRIPTION
**Pull request by nishaatr**
**note: change to be merged into 5.3.z**

1. Remove GPL license. Some dependencies are dual-licensed. Hazelcast uses the dependencies under one of the licenses. The other license is usually a copy-left license, which only confuses our users, so we exclude it
2. use SPDX identifiers for all 'included' licenses for clarity and reduce license merge lines proliferation

Fixes [HZ-2159](https://hazelcast.atlassian.net/browse/HZ-2159), [HZ-2774](https://hazelcast.atlassian.net/browse/HZ-2774)

[HZ-2159]: https://hazelcast.atlassian.net/browse/HZ-2159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HZ-2774]: https://hazelcast.atlassian.net/browse/HZ-2774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -521,37 +521,87 @@
...
+                                    Eclipse Distribution License - v 1.0 |
+                                    EDL 1.0
                                 </licenseMerge>
-                                <licenseMerge>BSD 2-Clause License|BSD-2-Clause|BSD 2-Clause license</licenseMerge>
-                                <licenseMerge>CC0|Public Domain, per Creative Commons CC0</licenseMerge>
                                 <licenseMerge>
-                                    CDDL|CDDL + GPLv2 with classpath exception|CDDL License|CDDL/GPLv2+CE
+                                    EPL-2.0 |
+                                    EPL 2.0 |
+                                    Eclipse Public License - Version 1.0
```
Comment (kwart) [inline pom.xml:600]: Wrong version.

**Review by kwart — CHANGES_REQUESTED** [review]
One typo, otherwise LGTM.

I see there are more license merge strings introduced. Do we want to have them in the configuration even when they are not used?

File: pom.xml (hunk 1 above)
Comment (nishaatr) [reply pom.xml:600]: Good question @kwart 
1. I kept everything from before as I didn't want to rock the boat so to speak (more testing will be needed). I think its possible some are unused tbh but to remove each one I would need to recompile hazelcast (and possibly hazelcast-enterprise as we do update some part of the license in there) for every removal which will take long. How about we keep as is and I raise a separate GH issue to clean-up the list?
2. good catch with the "EPL 2.0 | Eclipse Public License - Version 1.0". I will fix that shortly and push again

Comment (nishaatr) [reply pom.xml]: FYI...changed pushed https://github.com/hazelcast/hazelcast/pull/25150/commits/dddb016b9afd7f192745f1bfbd215bb28ef2368e

Comment (nishaatr) [reply pom.xml]: Pushed another update after further review comments from internal stakeholder

@kwart please approve if you are ok?

**Review by kwart — APPROVED** [review]

**Comment by kwart** [comment]
run-lab-run

**Pull request merged**